pkg/p2p/streamtest/streamtest.go

```diff
@@ -155,6 +161,10 @@ func (s *stream) Write(p []byte) (int, error) {
 	return s.in.Write(p)
 }
 
+func (s *stream) Headers() p2p.Headers {
+	return s.headers
```

Comment:
Should we add a method to set the headers for testing purposes maybe?

Reply:
Currently, this `stream` is not exported and not used as the mock for the stream, but a recorder. They are set in NewStream if protocol has Headler. I would not add SetHeaders method until it is needed, as it would imply exporting the stream in this package or creating a new interface with that method. If that is ok?

Reply:
Yes, u r right, sorry, I did not see that 🙈  Sounds good!